src/mlmc/sample_storage.py

```diff
@@ -223,30 +224,50 @@ def sample_pairs(self):
         levels_results = list(np.empty(len(np.max(self._results.keys()))))
 
         for level_id in self.get_level_ids():
-            results = self.sample_pairs_level(ChunkSpec(level_id))
+            results = self.sample_pairs_level(level_id=level_id)
             levels_results[level_id] = results
 
         return levels_results
 
-    def sample_pairs_level(self, chunk_spec):
+    def chunks(self, level_id=None, n_samples=None):
+        """
+        Create chunks generator
+        :param level_id: int, if not None return chunks for a given level
+        :return: generator
+        """
+        if level_id is not None:
```

Comment:
Can be moved to the StorageBase.

Reply:
I would create `@abstractmethod` in the parent class